detect/detector.py

```diff
@@ -57,19 +57,22 @@ def detect(self, det_iter, show_timer=False):
         list of detection results
         """
         num_images = det_iter._size
+        result = []
+        detections = []
         if not isinstance(det_iter, mx.io.PrefetchingIter):
             det_iter = mx.io.PrefetchingIter(det_iter)
         start = timer()
-        detections = self.mod.predict(det_iter).asnumpy()
+        for pred, i_batch, batch in self.mod.iter_predict(det_iter):
+            detections.append([output.asnumpy() for output in pred])
```

Comment:
how about 
```
for pred, _, _ in self.mod.iter_predict(det_iter):
    detections.append(pred[0].asnumpy())
```
since we don't accept other outputs if possible